Framework/DataHandling/src/LoadMuonNexusV2Helper.cpp

```diff
@@ -64,18 +70,22 @@ NXInt loadGoodFramesDataFromNexus(const NXEntry &entry,
 std::vector<detid_t>
 loadDetectorGroupingFromNexus(const NXEntry &entry,
                               const std::vector<detid_t> &detectorsLoaded,
-                              bool isFileMultiPeriod) {
-
+                              bool isFileMultiPeriod, const int periodNumber) {
+  // We cast the numLoadedDetectors to an int, which is the index type used for
+  // Nexus Data sets. As detectorsLoaded is derived from a nexus entry we
+  // can be certain we won't overflow the integer type.
+  int numLoadedDetectors = static_cast<int>(detectorsLoaded.size());
   std::vector<detid_t> grouping;
```

Comment:
`std::vector<detid_t> grouping(numLoadedDetectors)`

Will do the same as 
`std::vector<detid_t> grouping;`
`grouping.reserve(numLoadedDetectors)`

With the bonus the reserve can't get lost in the future